pkg/trace/stats/aggregation.go

```diff
@@ -80,15 +96,48 @@ func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregation
 			Synthetics: synthetics,
 		},
 	}
-	if enablePeerSvcAgg {
-		agg.PeerService = s.Meta[tagPeerService]
+	var peerTags []string
+	if spanKindIsClientOrProducer(agg.SpanKind) {
+		if enablePeerSvcAgg {
+			agg.PeerService = s.Meta[tagPeerService]
+		}
+		peerTags = getMatchingPeerTags(s, peerTagKeys)
+		agg.PeerTagsHash = peerTagsHash(peerTags)
 	}
-	return agg
+	return agg, peerTags
+}
+
+func getMatchingPeerTags(s *pb.Span, peerTagKeys []string) []string {
+	if len(peerTagKeys) == 0 {
+		return nil
+	}
+	pt := make([]string, 0)
+	for _, t := range peerTagKeys {
+		v, ok := s.Meta[t]
+		if ok {
+			pt = append(pt, t+":"+v)
+		}
+	}
+	if len(pt) == 0 {
+		return nil
+	}
+	return pt
+}
+
+func peerTagsHash(tags []string) uint64 {
+	if len(tags) == 0 {
+		return 0
+	}
+	h := fnv.New64a()
+	for _, t := range tags {
+		h.Write(stringToBytes(t))
```

Comment:
I believe the go compiler knows to allocate a byte array in the stack in this case because it isn't reused and you can just do `[]byte(t)`. Not sure it makes a difference either way but it might save having to reimplement and use this unsafe function.